dev/RestartAgent/main.cpp

```diff
@@ -0,0 +1,69 @@
+﻿#include "pch.h"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+
+using namespace AppModel::Identity;
+
+int APIENTRY wWinMain(HINSTANCE, HINSTANCE, LPWSTR, int)
+{
+    // Get the parent process handle id from the command-line.  We assume here
+    // that CreateProcess was called in a manner that inherited this handle to 
+    // the current process.
+
+    int argc{};
+    std::wstring cmdLine{ GetCommandLineW() };
+    wil::unique_hlocal_ptr<PWSTR[]> argv{ CommandLineToArgvW(cmdLine.c_str(), &argc) };
+
+    wil::unique_handle parentHandle;
+    auto handleArg{ argv[1] };
+    parentHandle.reset(reinterpret_cast<HANDLE>(_wtoi64(handleArg)));
+
+    // Determine the exact path of the calling process needing a restart.
+    auto callerPath = wil::QueryFullProcessImageNameW<wil::unique_cotaskmem_string>(parentHandle.get(), 0);
+
+    // Terminate the caller.  TerminateProcess is async, so we need to wait for the process to finish terminating.
+    THROW_IF_WIN32_BOOL_FALSE(TerminateProcess(parentHandle.get(), 0));
```

Comment:
App requesting restart is never given a chance to spin down, which could result in leaks (e.g. GDI handles). I noted the same concern in the design document PR.